s3stream/src/main/java/com/automq/stream/s3/operator/DeleteObjectsAccumulator.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2024, AutoMQ CO.,LTD.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+package com.automq.stream.s3.operator;
+
+import com.automq.stream.s3.metrics.TimerUtil;
+import com.automq.stream.s3.metrics.stats.S3OperationStats;
+import com.automq.stream.utils.FutureUtil;
+import com.automq.stream.utils.Threads;
+import com.google.common.util.concurrent.RateLimiter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class DeleteObjectsAccumulator implements Runnable {
```

Comment:
Is it possible to indirectly accumulate batches by limiting the number of Inflight Deletes? This way, even in cases where Deletes are relatively few, the latency of Deletes won't be compromised.

Reply:
For example, restrict the Delete concurrency to 10. If the Inflight does not exceed 10, it is freely released. If it exceeds 10, it queues up inside, waiting for the release in front before being taken out of the queue.

Reply:
current code support config no batch DeleteOptions to do this.
for the frontend request user can use this to skip time batch logic.

Reply:
In theory, higher-level code should not be aware of whether it needs to Batch.

Reply:
Because the underlying concurrency of S3 is also limited (default 50), in fact, for latency-sensitive APIs, limiting the number of inflights and accumulating batches through inflight restrictions is also better.

Reply:
got it. I'll change the code.